### PR TITLE
In memory searchparam bugfix

### PR DESCRIPTION
**Pull request by fil512**

### CODE REVIEW

**Comment by fil512** [comment]
@jamesagnew please merge if the tests pass

File: hapi-fhir-jpaserver-searchparam/src/main/java/ca/uhn/fhir/jpa/searchparam/extractor/ResourceIndexedSearchParams.java [hunk 1]
```diff
@@ -285,9 +285,9 @@ private boolean resourceIdMatches(ResourceLink theResourceLink, ReferenceParam t
 		}
 	}
 
-	private boolean resourceLinkMatches(String theResourceName, ResourceLink theResourceLink, String theParamName) {
+	private boolean resourceLinkMatches(String theResourceName, ResourceLink theResourceLink, String theParamName, String theParamPath) {
 		return theResourceLink.getTargetResource().getResourceType().equalsIgnoreCase(theParamName) ||
-			theResourceLink.getSourcePath().equalsIgnoreCase(theResourceName+"."+theParamName);
+			theResourceLink.getSourcePath().equalsIgnoreCase(theParamPath);
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-searchparam/src/main/java/ca/uhn/fhir/jpa/searchparam/extractor/ResourceIndexedSearchParams.java]: Ohhhhhhhh yeah. Boy, as soon as I see the actual fix it seems so obvious. Ha!

**Comment by jamesagnew** [comment]
@fil512 This looks great. I'll merge when the tests are done, but if I get distracted and forget please feel free to merge it yourself.

**Comment by jamesagnew** [comment]
Oh, I just realized one thing this needs: A changelog entry. Can you add in a followup commit? Merging this now.

**Pull request merged**